lib/Convos/Plugin/Auth/Header.pm

```diff
@@ -0,0 +1,90 @@
+package Convos::Plugin::Auth::Header;
+use Mojo::Base 'Convos::Plugin::Auth', -async_await;
+
+has admin_user         => sub { $ENV{CONVOS_ADMIN} // 'admin-email-not-set@example.com' };
+has x_user_header_name => sub { $ENV{CONVOS_AUTH_HEADER} || 'X-Authenticated-User' };
+
+sub register {
+  my ($self, $app, $config) = @_;
+  $self->SUPER::register($app, $config);
+  $self->admin_user($config->{admin_user})                 if defined $config->{admin_user};
+  $self->x_user_header_name($config->{x_user_header_name}) if $config->{x_user_header_name};
+  $app->helper('user.load_p' => sub { $self->_user_load_p(@_) });
+}
+
+async sub _user_load_p {
+  my ($self, $c) = @_;
+  my $core = $c->app->core;
+
+  my $header = $self->x_user_header_name;
+  my $email  = $c->req->headers->header($header);
+  return _giveup($c, debug => "Header $header is missing in request.") unless $email;
+
+  my $user = $email && $core->get_user({email => $email});
+  return $user if $user;
+
+  if (!$core->n_users and $self->admin_user and $email ne $self->admin_user) {
+    my $admin = $self->admin_user;
+    return _giveup($c, warn => "Cannot auto-register user, when admin $admin is not registered.");
+  }
+
+  $c->app->log->info("Auto-registering user $email from header $header");
+  return $core->user({email => $email})->save_p;
+}
+
+sub _giveup {
+  my ($c, $level, $msg) = @_;
+  $c->log->$level($msg);
+  return undef;
+}
+
+1;
+
+=encoding utf8
+
+=head1 NAME
+
+Convos::Plugin::Auth::Header - Authenticate users by verifying reverse proxy header
+
+=head1 SYNOPSIS
+
+  $ CONVOS_PLUGINS=Convos::Plugin::Auth::Header \
+    CONVOS_ADMIN=admin@example.com \
+    CONVOS_AUTH_HEADER=X-User \
+    ./script/convos daemon
+
+C<CONVOS_ADMIN> defaults to C<admin-email-not-set@example.com>, but can be set
+to the Convos admin user's email address to I<disallow> anyone else to register
+as a normal user before the admin has registered/logged in. Setting this
+enviroment variable to an empty string will make the first user the admin user.
+
+C<CONVOS_AUTH_HEADER> must contain the header name that will contain the email
```

Comment:
Can I use just a string instead of an email address for an user identifier? That would be very usefuld, except if an email is required for some other reason.

Reply:
Alright, thanks!